src/code.h

```diff
@@ -355,17 +355,18 @@ typedef Stat Expr;
 #define T_INT_EXPR              (FIRST_EXPR_TNUM+28)
 #define T_TRUE_EXPR             (FIRST_EXPR_TNUM+29)
 #define T_FALSE_EXPR            (FIRST_EXPR_TNUM+30)
-#define T_CHAR_EXPR             (FIRST_EXPR_TNUM+31)
-#define T_PERM_EXPR             (FIRST_EXPR_TNUM+32)
-#define T_PERM_CYCLE            (FIRST_EXPR_TNUM+33)
-#define T_LIST_EXPR             (FIRST_EXPR_TNUM+34)
-#define T_LIST_TILD_EXPR        (FIRST_EXPR_TNUM+35)
-#define T_RANGE_EXPR            (FIRST_EXPR_TNUM+36)
-#define T_STRING_EXPR           (FIRST_EXPR_TNUM+37)
-#define T_REC_EXPR              (FIRST_EXPR_TNUM+38)
-#define T_REC_TILD_EXPR         (FIRST_EXPR_TNUM+39)
-
-#define T_REFLVAR               (FIRST_EXPR_TNUM+40)
+#define T_TILDE_EXPR            (FIRST_EXPR_TNUM+31)
+#define T_CHAR_EXPR             (FIRST_EXPR_TNUM+32)
+#define T_PERM_EXPR             (FIRST_EXPR_TNUM+33)
+#define T_PERM_CYCLE            (FIRST_EXPR_TNUM+34)
+#define T_LIST_EXPR             (FIRST_EXPR_TNUM+35)
+#define T_LIST_TILD_EXPR        (FIRST_EXPR_TNUM+36)
+#define T_RANGE_EXPR            (FIRST_EXPR_TNUM+37)
+#define T_STRING_EXPR           (FIRST_EXPR_TNUM+38)
+#define T_REC_EXPR              (FIRST_EXPR_TNUM+39)
+#define T_REC_TILD_EXPR         (FIRST_EXPR_TNUM+40)
+
+#define T_REFLVAR               (FIRST_EXPR_TNUM+41)
```

Comment:
This is orthogonal to this PR, but: is there any reasons not to turn all these #defines into a single big enum, and then leave the numbering to the compiler? This looks like a prime example of what `enum` was made for... Inserting or removing new expr would result in a minimal diff.